### ISSUE TITLE
How to retain the background color of the selected items upon screen rotation?

### ISSUE THREAD

**Issue by ShahoodulHassan** [davideas/FlexibleAdapter#664]
HI,
I'm trying to implement headers and sections but facing a problem with action mode selection. When multi selection is turned on, I'm able to get the selected items' background color as set in accordance with [https://github.com/davideas/FlexibleAdapter/wiki/5.x-%7C-Utils#usage](url) but upon screen rotation, the background color is gone and none of the items **_appear_** as selected. However, the actionMode subtitle still shows the count of selected items.
Can you please help me to show the items as selected even after a screen rotation?

**Comment by ShahoodulHassan** [comment]
Ok what I have noticed is that on each screen rotation, the selected positions in `selectedItems` show position as **_position + adapter's item count_**.
So, for example, before screen rotation, position number 4 and 9 were selected and the total number of items displayed by the adapter were 20 (including headers). After screen rotation and after calling `mAdapter.onRestoreInstanceState()`, these positions will be changed to 24 and 29. On next rotation, 44 and 49 and so on and so forth.
Due to this issue, adapter doesnt treat any item as selected because the positions stored in `selectedItems` are not accessible by it...and therefore, the background color is not changed.

I must be doing something very very wrong here....so need help from this community. Should u need to look at any piece of code, pl ask for it.

Thanking in advance!

**Comment by ShahoodulHassan** [comment]
@davideas finally understood the issue....its the same as defined in [https://github.com/davideas/FlexibleAdapter/issues/581](url). Although, u have instructed in the wiki to look at this issue **Note: Pay attention to restore selection after adapter has been filled. More details in issue #581.** but I took two words **restore selection** as one word **restoreSelection**.  What I understood was that u instructed to ensure that adapter is filled before calling `restoreSelection` of `ActionModeHelper` while actually adapter MUST be filled before even `mAdapter.onRestoreInstanceState(savedInstanceState);` is called.
So, if u deem appropriate, pl rephrase the above mentioned note. I wasted 02 days on it but a rephrasing might save someone else's time.
